File: frontend/src/types/Exercise.ts
export type Exercise = {
    id: string;
    name: string;
    description?: string;
    image?: string;
    type: ExerciseType;
}

export enum ExerciseType {
    UPPER_BODY = "UPPER_BODY",
    LOWER_BODY = "LOWER_BODY",
}

File: backend/src/main/java/com/yuliiaskrypnyk/backend/model/exercise/Exercise.java
package com.yuliiaskrypnyk.backend.model.exercise;

import lombok.Builder;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Builder
@Document(collection = "exercises")
public record Exercise(
        @Id String id,
        String name,
        String description,
        String image) {
}
